File: 0001-two-sum/0001-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        # for i,v in enumerate(nums): [0,1,2,3,4]
        #     for j, value in enumerate(nums[i+1:]):
        #         if v + value == target:
        #             return [i, j+i+1]

        # 0,1 => 0,2 => 0,3 

        # dictionary = dict()
        # for i, v in enumerate(nums):
        #     if v in dictionary:
        #         dictionary[v].append(i)
        #     else:
        #         dictionary[v] = [i]
        

        # for i, v in enumerate(nums):
        #     if target - v in dictionary:
        #         for value in dictionary[target-v]:
        #             if value != i:
        #                 return [i, value] 
        dictionary = {}
        for i, v in enumerate(nums):
            need = target - v
            if need in dictionary:
                return [dictionary[need], i]
            dictionary[v] = i
            
        

File: LeetCode/Easy/0001-two-sum/0001-two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer, Integer> map = new HashMap<>();
        int[] result = new int[2];
        for (int i = 0; i < nums.length; ++i){
            int need = target - nums[i];
            if (map.containsKey(need)){
                result[0] = map.get(need);
                result[1] = i;
                break;
            }
            map.put(nums[i], i);
        }
        return result;
    }
}